### PR TITLE
docs: Edit catalog API docs

### PR DESCRIPTION
**Pull request by pierwill**
Makes edits to the catalog API docs for clarity, formatting, consistency, and readability.

### CODE REVIEW

File: iox_catalog/src/interface.rs [hunk 1]
```diff
@@ -162,24 +163,28 @@ pub(crate) mod sealed {
     }
 }
 
-/// transaction of a [`Catalog`].
+/// Transaction in a [`Catalog`] (similar to a database transaction).
```
Comment (pierwill) [inline iox_catalog/src/interface.rs:166]: "In" makes the most sense to me, as I understand it. 

Maybe "for a catalog"? 🤔

File: iox_catalog/src/interface.rs [hunk 2]
```diff
@@ -162,24 +163,28 @@ pub(crate) mod sealed {
...
 ///
-/// A transaction provides a consistent view on data and stages writes (this normally maps to a database transaction).
-/// Repositories can cheaply be borrowed from it. To finalize a transaction, call [commit](Self::commit) or [abort](Self::abort).
+/// A transaction provides a consistent view on data and stages writes.
+/// To finalize a transaction, call [commit](Self::commit) or [abort](Self::abort).
 ///
-/// Note that after any method in this transaction (including all repositories derived from it) returned an error, the
+/// Repositories can cheaply be borrowed from it.
```
Comment (pierwill) [inline iox_catalog/src/interface.rs:171]: Maybe add an example here?

File: iox_catalog/src/interface.rs (hunk 1 above)
Comment (alamb) [reply iox_catalog/src/interface.rs:166]: I think `Transaction in` makes sense ( though I think `of` is probably also technically accurate)

**Review by alamb — APPROVED** [review]
Looks like a great set of improvements to me -- thanks @pierwill !

**Pull request merged**